File: 2025/Company/Company.Web/Modules/Company/Currency/CurrencyColumns.cs
namespace Company.Company.Columns;

[ColumnsScript("Company.Currency")]
[BasedOnRow(typeof(CurrencyRow), CheckNames = true)]
public class CurrencyColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int CurrencyId { get; set; }
    [EditLink]
    public string CurrencyCode { get; set; }
    public string Name { get; set; }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/CurrencyColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CurrencyRow } from "./CurrencyRow";

export interface CurrencyColumns {
    CurrencyId: Column<CurrencyRow>;
    CurrencyCode: Column<CurrencyRow>;
    Name: Column<CurrencyRow>;
}

export class CurrencyColumns extends ColumnsBase<CurrencyRow> {
    static readonly columnsKey = 'Company.Currency';
    static readonly Fields = fieldsProxy<CurrencyColumns>();
}